Cheetah/Template.py

```diff
@@ -44,32 +45,33 @@ def __init__(
             is a function which takes a single argument (the contents of a
             template variable) and may perform some output filtering.
         """
-        if searchList:
-            for namespace in searchList:
-                if (
-                        isinstance(namespace, dict) and
-                        self.Reserved_SearchList & set(namespace)
-                ):
-                    raise AssertionError(
-                        'The following keys are members of the Template class '
-                        'and will result in NameMapper collisions!\n'
-                        '  > {0} \n'
-                        "Please change the key's name.".format(
-                            ', '.join(
-                                self.Reserved_SearchList &
-                                set(namespace)
-                            )
+        if namespace:
+            if (
+                    isinstance(namespace, dict) and
+                    self.Reserved_SearchList & set(namespace)
+            ):
+                raise AssertionError(
+                    'The following keys are members of the Template class '
+                    'and will result in NameMapper collisions!\n'
+                    '  > {0} \n'
+                    "Please change the key's name.".format(
+                        ', '.join(
+                            self.Reserved_SearchList &
+                            set(namespace)
                         )
                     )
-
-        if searchList is not None and not isinstance(searchList, (list, tuple)):
-            raise AssertionError(
-                'Expected searchList to be `None`, `list`, or `tuple` '
-                'but got {0}'.format(type(searchList))
+                )
+
+        if (
+                namespace is not None and
+                not isinstance(namespace, collections.Mapping)
+        ):
+            raise TypeError(
+                '`namespace` must be `Mapping` but got {0!r}'.format(namespace)
             )
 
         # create our own searchList
-        self._CHEETAH__searchList = [self] + list(searchList or [])
+        self._CHEETAH__searchList = [self, namespace or {}]
```

Comment:
This isn't ideal since I can't pass in an empty dict and keep a reference to it.
I'd rather see the conventional `if namespace is None: namespace = {}`.

Also is this the right time to replace `_CHEETAH__searchList` with `_CHEETAH__namespace`?